extension/src/popup/views/RecoverAccount/index.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { Field, Formik, FieldProps } from "formik";
+import { object as YupObject } from "yup";
+
+import { FullscreenStyle } from "popup/components/FullscreenStyle";
+import { ROUTES } from "popup/constants/routes";
+import { navigateTo } from "popup/helpers/navigate";
+import {
+  password as passwordValidator,
+  confirmPassword as confirmPasswordValidator,
+  termsOfUse as termsofUseValidator,
+  mnemonicPhrase as mnemonicPhraseValidator,
+} from "popup/helpers/validators";
+import {
+  authErrorSelector,
+  publicKeySelector,
+  recoverAccount,
+} from "popup/ducks/accountServices";
+import { FormRow, Form } from "popup/basics/Forms";
+import { Header } from "popup/components/Header";
+import { PasswordRequirements } from "popup/components/PasswordRequirements";
+
+import { Input, Button, Checkbox, TextLink } from "@stellar/design-system";
+
+import "./styles.scss";
+
+export const RecoverAccount = () => {
+  const publicKey = useSelector(publicKeySelector);
+  const authError = useSelector(authErrorSelector);
+
+  interface FormValues {
+    password: string;
+    confirmPassword: string;
+    mnemonicPhrase: string;
+    termsOfUse: boolean;
+  }
+
+  const initialValues: FormValues = {
+    password: "",
+    confirmPassword: "",
+    mnemonicPhrase: "",
+    termsOfUse: false,
+  };
+
+  const RecoverAccountSchema = YupObject().shape({
+    mnemonicPhrase: mnemonicPhraseValidator,
+    password: passwordValidator,
+    confirmPassword: confirmPasswordValidator,
+    termsOfUse: termsofUseValidator,
+  });
+
+  const dispatch = useDispatch();
+
+  const handleSubmit = async (values: FormValues) => {
+    const { password, mnemonicPhrase } = values;
+
+    await dispatch(
+      recoverAccount({
+        password,
+        mnemonicPhrase: mnemonicPhrase.trim(),
+      }),
+    );
+  };
+
+  useEffect(() => {
+    if (publicKey) {
+      navigateTo(ROUTES.recoverAccountSuccess);
+      window.close();
+    }
+  }, [publicKey]);
+
+  return (
+    <>
+      <Header />
+      <FullscreenStyle />
+      <Formik
+        initialValues={initialValues}
+        onSubmit={handleSubmit}
+        validationSchema={RecoverAccountSchema}
+      >
+        {({ dirty, touched, isSubmitting, isValid, errors }) => (
+          <Form>
+            <div className="RecoverAccount__screen">
+              <div className="RecoverAccount__half-screen">
+                <div className="RecoverAccount__header">
+                  Import wallet from recovery phrase
+                </div>
+                <FormRow>
+                  <div className="RecoverAccount__mnemonic-input">
+                    <Field name="mnemonicPhrase">
+                      {({ field }: FieldProps) => (
+                        <>
+                          <textarea
```

Comment:
@quietbits thoughts on this textarea input using the Card class? Should we add to SDS?

Reply:
Agreed that we should add textarea component to SDS. We'll need to check with the Design about styling, though. We haven't used this element anywhere else, so not sure if the style should always be like this. Good workaround in this case! :+1: